frontend/src/sagas/schedule.js

```diff
@@ -0,0 +1,441 @@
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setScheduleLoading(false));
+    }
+}
+
+export function* getScheduleItems() {
+    try {
+        const response = yield call(axiosCall, CURRENT_SEMESTER_URL);
+        yield put(setCurrentSemester(response.data));
+        yield call(showBusyRooms, response.data.id);
+        yield call(getScheduleItemsBySemester, { semesterId: response.data.id });
+    } catch (error) {
+        const message = i18n.t(NO_CURRENT_SEMESTER_ERROR);
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
+    }
+}
+
+export function* addItemsToSchedule({ item }) {
+    try {
+        yield call(axiosCall, SCHEDULE_ITEMS_URL, 'POST', item);
+        yield call(getScheduleItems);
+    } catch (error) {
+        const message = error.response
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+    }
+}
+
+export function* checkAvailabilityChangeRoomSchedule({ item }) {
+    const { periodId, dayOfWeek, evenOdd, semesterId } = item;
+    const requestUrl = `${ROOMS_AVAILABILITY}?classId=${periodId}&dayOfWeek=${dayOfWeek}&evenOdd=${evenOdd}&semesterId=${semesterId}`;
+    try {
+        const response = yield call(axiosCall, requestUrl);
+        yield put(
+            checkAvailabilitySchedule({
+                classSuitsToTeacher: true,
+                teacherAvailable: true,
+                rooms: response.data,
+            }),
+        );
+        yield put(setLoading(false));
+    } catch (error) {
+        const message = error.response
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
+    }
+}
+
+export function* checkScheduleItemAvailability({ item }) {
+    const { periodId, dayOfWeek, evenOdd, lessonId, semesterId } = item;
+    const requestUrl = `${SCHEDULE_CHECK_AVAILABILITY_URL}?classId=${periodId}&dayOfWeek=${dayOfWeek}&evenOdd=${evenOdd}&lessonId=${lessonId}&semesterId=${semesterId}`;
+    try {
+        const response = yield call(axiosCall, requestUrl);
+        yield put(checkAvailabilitySchedule(response.data));
+        yield put(setLoading(false));
+    } catch (error) {
+        const message = error.response
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
```

Comment:
@Akr1s 
maybe to use finally for yield put(setLoading(false));

Reply:
changed in another branch